src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -548,12 +495,75 @@ internal static Matrix3x2 GetTransform(UIElement from, UIElement to)
 			return matrix;
 		}
 
+		/// <summary>
+		/// Applies to the given matrix the transformation needed to convert from parent to local element coordinates space.
+		/// </summary>
+		/// <param name="matrix">The matrix into which the layout constraints should be written</param>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		internal void ApplyLayoutTransform(ref Matrix3x2 matrix)
+		{
+			// This is equivalent to matrix * Matrix3x2.CreateTranslation(layout.X, layout.Y);
+			var layoutSlot = LayoutSlotWithMarginsAndAlignments;
+			matrix.M31 += (float)layoutSlot.X;
+			matrix.M32 += (float)layoutSlot.Y;
+		}
+
+		/// <summary>
+		/// Applies to the given matrix the <see cref="RenderTransform"/>.
+		/// </summary>
+		/// <param name="matrix">The matrix into which the render transform should be written</param>
+		/// <param name="ignoreOrigin">Indicates if the <see cref="RenderTransformOrigin"/> should be ignored or not.</param>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		internal void ApplyRenderTransform(ref Matrix3x2 matrix, bool ignoreOrigin = false)
+		{
+			if (RenderTransform is { } transform)
+			{
+				var transformMatrix = transform.MatrixCore;
+				if (!ignoreOrigin)
+				{
+					transformMatrix = transformMatrix.CenterOn(RenderTransformOrigin, LayoutSlotWithMarginsAndAlignments.Size);
+				}
+
+				matrix *= transformMatrix;
+			}
+		}
+
+		/// <summary>
+		/// Applies to the given matrix the constrains specific to the current element (like ScrollOffsets).
+		/// </summary>
+		/// <param name="matrix">The matrix into which the transformations should be written</param>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		internal void ApplyElementCustomTransform(ref Matrix3x2 matrix)
+		{
+#if !__MACOS__ // On macOS the SCP is using RenderTransforms for scrolling and zooming which has already been included.
+			if (this is ScrollViewer sv)
+			{
+				// Scroll offsets are handled at SCP level using the IsScrollPort
+				// TODO: ZoomFactor should also be handled at the SCP level!
```

Comment:
```suggestion
				// Scroll offsets are handled at the SCP level using the IsScrollPort
				// TODO: ZoomFactor should also be handled at the SCP level!
```

Do you have an issue or task that you can link for this TODO ;) ?